var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -434,6 +434,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("boost+graph+math+exception+stacktrace", when="@:13.4.0 +stk")
     depends_on("cgns", when="+exodus")
     depends_on("cmake@3.23:", type="build", when="@14.0.0:")
+    conflicts("cmake@3.31:", when="@14.4.0")
```

Comment:
I would suggest adding a [forward compat bound](https://spack.readthedocs.io/en/v0.23.0/packaging_guide.html#specifying-backward-and-forward-compatibility) like this:

```suggestion
    depends_on("cmake@:3.30", type="build", when="@:14.3.0")
```

which is kinda symmetric with the backward compat bound a line above.

Cause it sounds unlikely that older trilinos versions *do* work with newer cmake 3.31?

Reply:
I wasn't sure what range of versions were incompatible - hence limited my initial change to the one version I could verify.